.script/tests/detectionTemplateStructureValidation/DetectionTemplateStructureValidationTests.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System.IO;
+using System.Linq;
+using FluentAssertions;
+using Microsoft.Azure.Sentinel.Analytics.Management.AnalyticsTemplatesService.Interface.Model;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Xunit;
+using YamlDotNet.Serialization;
+
+namespace Kqlvalidations.Tests
+{
+    public class DetectionTemplateStructureValidationTests
+    {
+        private static readonly string DetectionPath = DetectionsYamlFilesTestData.GetDetectionPath();
+
+        [Theory]
+        [ClassData(typeof(DetectionsYamlFilesTestData))]
+        public void Validate_DetectionTemplates_HaveValidTemplateStructure(string detectionsYamlFileName)
+        {
+            var detectionsYamlFile = Directory.GetFiles(DetectionPath, detectionsYamlFileName, SearchOption.AllDirectories).Single();
+            var yaml = File.ReadAllText(detectionsYamlFile);
+
+            //we ignore known issues (in progress)
+            foreach (var templateToSkip in TemplatesToSkipValidationReader.WhiteListTemplateIds)
+            {
+                if (yaml.Contains(templateToSkip) || detectionsYamlFile.Contains(templateToSkip))
+                {
+                    return;
+                }
+            }
+
+            var jObj = JObject.Parse(ConvertYamlToJson(yaml));
+
+            var exception = Record.Exception(() => jObj.ToObject<ScheduledTemplateInternalModel>());
```

Comment:
consider using JSON Schama validation instead of if serilize is throwing exception - you can generate the schema from type - https://www.newtonsoft.com/jsonschema/help/html/GeneratingSchemas.htm
maybe store the schema only and then no need to have all the data model
https://www.newtonsoft.com/json/help/html/JsonSchema.htm WDYT

Reply:
next phase?